invenio_pidstore/__init__.py

```diff
@@ -22,7 +22,247 @@
+
+or alternatively simply assign the object while you create the persistent
+identifier (and also set the correct status immediately):
+
+>>> pid = PersistentIdentifier.create('recid', '1',
+...     object_type='rec', object_uuid=uuid.uuid4(),
+...     status=PIDStatus.REGISTERED)
+>>> pid.has_object()
+True
+
+Note, that once an object has been assigned you cannot assign a new object
+unless you use the overwrite parameter:
+
+>>> pid.assign('rec', uuid.uuid4())
+PIDObjectAlreadyAssigned
+>>> pid.assign('rec', uuid.uuid4(), overwrite=True)
+True
+
+Modifying status
+~~~~~~~~~~~~~~~~
+You can modify the state of a persistent identifier:
+
+>>> pid = PersistentIdentifier.create('doi', '10.1234/bar')
+>>> pid.reserve()
+True
+>>> pid.register()
+True
+>>> pid.delete()
+True
+
+Note that if you delete a persistent identifier in state *new* it will be
+completely removed from the database. Also, certain state changes are not
+allowed (e.g. here trying to reserve a deleted persistent identifier):
+
+>>> pid.reserve()
+PIDInvalidAction
+
+Redirecting
+~~~~~~~~~~~
+You can redirect a registered  persistent identifier to another persistent
+identifier. This is can be useful in cases where you merge two underlying
+records, or you have multiple identifier schemes but one is preferred over
+another.
+
+>>> pid = PersistentIdentifier.create('recid', '3',
+...     status=PIDStatus.REGISTERED)
+>>> pid.redirect('1', pid_type='recid')
+True
+
+This allows you easily retrieve the persistent identifier being redirected to:
+
+>>> new_pid = pid.get_redirect()
+
+Resolver
+--------
+The resolver is responsible for retrieving an internal object for a given
+persistent identifier.
+
+Following is an example for creating a record resolver:
+
+>>> from invenio.pidstore.resolver import Resolver
+>>> resolver = Resolver(pid_type='recid', object_type='rec', getter=Record.get)
+
+Above creates a resolve that will translate ``recid`` persistent identifiers
+to internal records. The getter argument must be a callable that takes one
+argument which is the persistent identifiers object UUID.
+
+You can now use the resolver to retrieve records:
+
+>>> record = resolver.resolve('1')
+
+The resolver will only resolve **registered** persistent identifiers. Any
+non-resolvable value will raise an exception. E.g. trying to resolve a deleted
+persistent will throw a :py:exc:`PIDDeletedError` exception:
+
+>>> resolver.resolve('2')
+
+Providers
+---------
+Providers adds extra functionality persistent identifiers. Use cases for this
+includes automatically creating the persistent identifier or retrieving the
+persistent identifier from an external service.
+
+PIDStore comes by default with a :py:class:`RecordIdProvider` which will
+create Invenio legacy integer record identifiers:
+
+>>> from invenio_pidstore.providers.recordid import RecordIdProvider
+>>> provider = RecordIdProvider.create()
+>>> provider.pid.pid_type
+'recid'
+>>> provider.pid.pid_value
+'1'
```

Comment:
I see that even if it is an `integer` it is still returned as a `str`. Is this a requirement for providers?